### ISSUE TITLE
Add ability to ignore packages

### ISSUE THREAD

**Issue by mandre** [joelanford/go-apidiff#35]
It could be useful to allow user to specify a list of packages to ignore when checking for API compatibility, when one does not care about a subset of packages such as tests.

**Comment by joelanford** [comment]
I'm not sure this is something we should support or encourage (see https://github.com/joelanford/go-apidiff/issues/14#issuecomment-1086866074).

I would recommend moving test utilities to an internal tree if you do not want users of your library to use them.

**Comment by mandre** [comment]
I understand that the `internal` directory is a go convention and moving files around would solve the issue for us, however not everyone wants to adopt this convention which causes users to implement workarounds like https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2567.

From my point of view, it's still useful to allow users to ignore specific paths.

**Comment by joelanford** [comment]
> I understand that the internal directory is a go convention

It is more than a convention. Dave Cheney has a succinct blog post about it here: https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface

TL;DR: `internal` packages are enforced to be internal by the go compiler. Conversely, non-internal packages are external and can be imported by your users.

As such, any breaking change in a non-internal package is, by definition, a breaking change to the API surface of the go module. 

The goal of go-apidiff is to report on whether a change could have the potential to break any importers of the changed packages.

> not everyone wants to adopt this convention

What is holding folks back from adopting use of `internal` trees for non-public APIs?

**Comment by EmilienM** [comment]
@joelanford what about this use case: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2096

v1beta1 package is stable, but not v1alpha1. I still want to run apidiff against v1beta1.

**Comment by deefdragon** [comment]
As much as supporting the explicit standard set out by the go compiler is nice, I think the [Robustness principal](https://en.wikipedia.org/wiki/Robustness_principle) needs to be applied here. 

Go is too large a language for EVERYONE to be following the standards explicitly (Ive worked at 2 large companies that openly flaunted some of them), and apidiff should probably be liberal in accepting the working patterns of others. IE allow filtering for the cases that don't follow the "Everything not in internal can be imported by somewhere else"

Also, while the go compiler _is_ the default, it's not the ONLY way to compile golang. IIRC, there are atleast two other compilers, one of which adopts most go _compiler_ standards, the other of which only adopts the go _language_ standards.
